invertedai/common.py

```diff
@@ -141,22 +141,34 @@ class AgentAttributes(BaseModel):
     AgentState
     """
 
-    length: float  #: Longitudinal extent of the agent, in meters.
-    width: float  #: Lateral extent of the agent, in meters.
+    length: Optional[float] = None  #: Longitudinal extent of the agent, in meters.
+    width: Optional[float] = None  #: Lateral extent of the agent, in meters.
     #: Distance from the agent's center to its rear axis in meters. Determines motion constraints.
-    rear_axis_offset: float
+    rear_axis_offset: Optional[float] = None
+    agent_type: Optional[str] = 'car'  #: Type of agent, default to car, can also be pedestrian
```

Comment:
```suggestion
    agent_type: Optional[str] = 'car'  #: Valid types are those in `AgentType`, but we use `str` here for extensibility.
```
I actually liked having an `AgentType` enum, even though we use `str` here. We need to have an authoritative list of valid agent types somewhere, and it only makes sense that we would expose it programmatically. Let's just be clear that this enum is extensible and we expect it to grow in future releases.